precompiles/staking/types.go

```diff
@@ -49,6 +56,110 @@ type EventCancelUnbonding struct {
+	if len(args) != 7 {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 7, len(args))
+	}
+
+	descriptionJson, err := json.Marshal(args[0])
+	if err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to marshal description to json: %v", err)
+	}
+	var description Description
+	if err := json.Unmarshal(descriptionJson, &description); err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to unmarshal json to description: %v", err)
+	}
+
+	commissionJson, err := json.Marshal(args[1])
+	if err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to marshal commission to json: %v", err)
+	}
+	var commission Commission
+	if err := json.Unmarshal(commissionJson, &commission); err != nil {
+		return nil, common.Address{}, fmt.Errorf("failed to unmarshal json to commission: %v", err)
+	}
+
+	minSelfDelegation, ok := args[2].(*big.Int)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidAmount, args[2])
+	}
+
+	delegatorAddress, ok := args[3].(common.Address)
+	if !ok || delegatorAddress == (common.Address{}) {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidDelegator, args[3])
+	}
+
+	validatorAddress, ok := args[4].(string)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidType, "validatorAddress", "string", args[4])
+	}
+
+	// use cli `evmosd tendermint show-validator` get pubkey
+	pubkeyBase64Str, ok := args[5].(string)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidType, "pubkey", "string", args[5])
+	}
+	pubkeyBytes, err := base64.StdEncoding.DecodeString(pubkeyBase64Str)
+	if err != nil {
+		return nil, common.Address{}, err
+	}
+	var ed25519pk cryptotypes.PubKey = &ed25519.PubKey{Key: pubkeyBytes}
+	pubkey, err := codectypes.NewAnyWithValue(ed25519pk)
+	if err != nil {
+		return nil, common.Address{}, err
+	}
+
+	value, ok := args[6].(*big.Int)
+	if !ok {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidAmount, args[6])
+	}
+
+	msg := &stakingtypes.MsgCreateValidator{
+		Description: stakingtypes.Description{
+			Moniker:         description.Moniker,
+			Identity:        description.Identity,
+			Website:         description.Website,
+			SecurityContact: description.SecurityContact,
+			Details:         description.Details,
+		},
+		Commission: stakingtypes.CommissionRates{
+			Rate:          sdk.NewDecFromBigIntWithPrec(commission.Rate, math.LegacyPrecision),
+			MaxRate:       sdk.NewDecFromBigIntWithPrec(commission.MaxRate, math.LegacyPrecision),
+			MaxChangeRate: sdk.NewDecFromBigIntWithPrec(commission.MaxChangeRate, math.LegacyPrecision),
+		},
+		MinSelfDelegation: sdk.NewIntFromBigInt(minSelfDelegation),
+		DelegatorAddress:  sdk.AccAddress(delegatorAddress.Bytes()).String(),
+		ValidatorAddress:  validatorAddress,
+		Pubkey:            pubkey,
+		Value:             sdk.NewCoin(denom, sdk.NewIntFromBigInt(value)),
```

Comment:
```suggestion
		Value:             sdk.Coin{Denom: denom, Amount: sdk.NewIntFromBigInt(value)},
```

Reply:
it's validated in validate basic

Reply:
done

Reply:
Well, to be honest, I don’t know a more reasonable reason why **sdk.NewCoin(denom, sdk.NewIntFromBigInt(value))** is replaced by **sdk.Coin{Denom: denom, Amount: sdk.NewIntFromBigInt(value)}**. I think both are fine.